### PR TITLE
Added support for reduced accuracy detection on iOS 14

### PR DESCRIPTION
**Pull request by leonluc-dev**
### Description of Change ###

iOS 14 added the ability for users to grant location permission with reduced accuracy.
In such cases the requested accuracy is ignored and the coarse location is returned.

This PR adds:

* A property indicating if the retrieved location has reduced accuracy (if true, the accuracy was intentionally reduced by the user through permission settings)
* A property to ask the user for temporary full accuracy authorization in the location request

**Platform notes:** 
* The `ReducedAccuracy` property always returns false on non-iOS platforms
* As [described in the Apple documentation](https://developer.apple.com/documentation/corelocation/cllocationmanager/3600217-requesttemporaryfullaccuracyauth); on iOS the temporary full accuracy request requires a purpose description to be added to the `NSLocationTemporaryUsageDescriptionDictionary` in the app's plist file. The key `XamarinEssentialsFullAccuracyUsageDescription` was used for this purpose.

### Bugs Fixed ###

- Fixes issue #1523

### API Changes ###

Added: 
 
- `bool Location.ReducedAccuracy { get; set; }`
- `bool GeolocationRequest.RequestFullAccuracy { get; set; }`

### PR Checklist ###

- [X] Has tests (if omitted, state reason in description)
- [X] Has samples (if omitted, state reason in description)
- [X] Rebased on top of `main` at time of PR
- [X] Changes adhere to coding standard
- [ ] Updated documentation ([see walkthrough](https://github.com/xamarin/Essentials/wiki/Documenting-your-code-with-mdoc))

### CODE REVIEW

File: Xamarin.Essentials/Geolocation/Geolocation.ios.macos.cs [hunk 1]
```diff
@@ -19,7 +19,14 @@ static async Task<Location> PlatformLastKnownLocationAsync()
             var manager = new CLLocationManager();
             var location = manager.Location;
 
-            return location?.ToLocation();
+            var reducedAccuracy = false;
+#if __IOS__
+            if (UIKit.UIDevice.CurrentDevice.CheckSystemVersion(14, 0))
```
Comment (jamesmontemagno) [inline Xamarin.Essentials/Geolocation/Geolocation.ios.macos.cs:24]: You can use our Platform helpers or deviceinfo. We like to re-use our logic.

**Review by jamesmontemagno — CHANGES_REQUESTED** [review]
I think beyond my feedback seems fine. XML docs need to be updated, but besides that is good.

**Comment by leonluc-dev** [comment]
@jamesmontemagno I've adjusted the version check to use the DeviceInfo API as suggested

File: Xamarin.Essentials/Geolocation/Geolocation.ios.macos.cs [hunk 2]
```diff
@@ -19,7 +19,14 @@ static async Task<Location> PlatformLastKnownLocationAsync()
             var manager = new CLLocationManager();
             var location = manager.Location;
 
-            return location?.ToLocation();
+            var reducedAccuracy = false;
+#if __IOS__
+            if (DeviceInfo.Version.Major >= 14)
```
Comment (jamesmontemagno) [inline Xamarin.Essentials/Geolocation/Geolocation.ios.macos.cs:24]:
```suggestion
            if (Platform.HasOSVersion(14,0))
```

**Review by jamesmontemagno — CHANGES_REQUESTED** [review]

File: Xamarin.Essentials/Geolocation/Geolocation.ios.macos.cs [hunk 3]
```diff
@@ -51,9 +58,22 @@ static async Task<Location> PlatformLocationAsync(GeolocationRequest request, Ca
 
             manager.StartUpdatingLocation();
 
+            var reducedAccuracy = false;
+#if __IOS__
+            if (DeviceInfo.Version.Major >= 14)
```
Comment (jamesmontemagno) [inline Xamarin.Essentials/Geolocation/Geolocation.ios.macos.cs:63]:
```suggestion
            if (Platform.HasOSVersion(14,0))
```

**Comment by leonluc-dev** [comment]
@jamesmontemagno The suggested changes have been applied

**Review by jamesmontemagno — APPROVED** [review]

**Review by 111mma — APPROVED** [review]

**Comment by jfversluis** [comment]
Not sure if we should still pull this in for Xamarin.Essentials since it's adding APIs but at the same time it also supports newer iOS functionalities. THoughts @jamesmontemagno ?

**Comment by jamesmontemagno** [comment]
I'm relatively fine with it if we do a 1.8.0

**Comment by mina5500** [comment]
@jamesmontemagno @jfversluis 
is it released in Xamarin.Essentials version 1.8.0 ?

**Review by jfversluis — APPROVED** [review]
Looks like this exact code is running in .NET MAUI without too many problems and we have updated Xamarin.Essentials with another major version, so let's do it.

**Pull request merged**